assets/common/src/batch_deploy.py

```diff
@@ -0,0 +1,383 @@
+    )
+    parser.add_argument(
+        "--max_instances",
+        type=int,
+        default=DEFAULT_MAX_INSTANCES,
+        help="Maximum number of instances of the compute cluster",
+    )
+    parser.add_argument(
+        "--idle_time_before_scale_down",
+        type=int,
+        default=DEFAULT_IDLE_TIME_BEFORE_SCALE_DOWN,
+        help="Node Idle Time before scaling down amlCompute",
+    )
+    parser.add_argument(
+        "--output_file_name",
+        type=str,
+        default=DEFAULT_OUTPUT_FILE_NAME,
+        help="Name of the batch scoring output file",
+    )
+    parser.add_argument(
+        "--max_concurrency_per_instance",
+        type=int,
+        default=DEFAULT_MAX_CONCURRENCY_PER_INSTANCE,
+        help="The maximum number of parallel scoring_script runs per instance",
+    )
+    parser.add_argument(
+        "--error_threshold",
+        type=int,
+        default=DEFAULT_ERROR_THRESHOLD,
+        help="The number of file failures that should be ignored",
+    )
+    parser.add_argument(
+        "--max_retries",
+        type=int,
+        default=DEFAULT_MAX_RETRIES,
+        help="The maximum number of retries for a failed or timed-out mini batch",
+    )
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        default=DEFAULT_TIMEOUT,
+        help="The timeout in seconds for scoring a single mini batch.",
+    )
+    parser.add_argument(
+        "--logging_level",
+        type=str,
+        default=DEFAULT_LOGGING_LEVEL,
+        help="The log verbosity level",
+    )
+    parser.add_argument(
+        "--mini_batch_size",
+        type=int,
+        default=DEFAULT_MINI_BATCH_SIZE,
+        help="The number of files the code_configuration.scoring_script can process in one run() call",
+    )
+    parser.add_argument(
+        "--instance_count",
+        type=int,
+        help="The number of nodes to use for each batch scoring job",
+        default=DEFAULT_INSTANCE_COUNT,
+        choices=range(1, MAX_INSTANCE_COUNT),
+    )
+    parser.add_argument(
+        "--model_deployment_details",
+        type=str,
+        help="Json file to which deployment details will be written",
+    )
+    # parse args
+    args = parser.parse_args()
+    logger.info(f"Args received {args}")
+    print("args received ", args)
+
+    return args
+
+
+def invoke_endpoint_job(ml_client, endpoint, type, args):
+    """Invoke a job using the endpoint."""
+    print(f"Invoking inference with {type} test payload ...")
+    try:
+        if type == "uri_folder":
+            path = args.inference_payload_folder
+        else:
+            path = args.inference_payload_file
+        input = Input(type=type, path=rf"{path}")
+
+        job = ml_client.batch_endpoints.invoke(
+            endpoint_name=endpoint.name, input=input
+        )
+
+        ml_client.jobs.stream(job.name)
+
+        scoring_job = list(ml_client.jobs.list(parent_job_name=job.name))[0]
+
+        ml_client.jobs.download(
```

Comment:
where are we downloading file to?

Reply:
Its downloaded in the default datastore

Reply:
Could we add `batch_job_output_folder` and set download_path for ml_client.jobs.download to that in the script?

Reply:
Have added the same